tracing-honeycomb/src/honeycomb.rs

```diff
@@ -34,6 +35,14 @@ impl HoneycombTelemetry {
             eprintln!("error sending event to honeycomb, {:?}", err);
         }
     }
+
+    fn should_report(&self, trace_id: TraceId) -> bool {
+        if self.sample_rate <= 1 {
+            return true;
+        }
+
+        trace_id.0 as usize % self.sample_rate == 0
```

Comment:
This is clever but I would have expected the sample rate to be a `0` to `1` kind of range - is this sort of sample rate common in other tracing libraries?

Reply:
I would also expect a float of some sort here - maybe an f32, with a 0 to 1 range

Reply:
Also, it looks like you're using sample_rate <= 1 to signify 'sample everything' - how do you feel about making this an `Option<SampleRate>` instead, where `type SampleRate = u32` (or `f32` or w/e)